contracts/StakeManagerV2.sol

```diff
@@ -0,0 +1,343 @@
+        BatchWithdrawalRequest storage batchRequest = batchWithdrawalRequests[firstUnbondingBatchIndex];
+        if (batchRequest.unlockTime > block.timestamp) revert Unbonding();
+
+        batchRequest.isClaimable = true;
+        firstUnbondingBatchIndex++;
+        STAKE_HUB.claim(batchRequest.operator, 1); // claims 1 request
+
+        emit CompletedBatchUndelegation(batchRequest.operator, batchRequest.amountInBnb);
+    }
+
+    /// @notice Claim the BNB from a withdrawal request.
+    /// @param _idx The index of the withdrawal request.
+    /// @return The amount of BNB claimed.
+    function claimWithdrawal(uint256 _idx) external override whenNotPaused nonReentrant returns (uint256) {
+        if (userRequests[msg.sender].length == 0) revert NoWithdrawalRequests();
+        if (_idx >= userRequests[msg.sender].length) revert InvalidIndex();
+
+        WithdrawalRequest storage request = withdrawalRequests[userRequests[msg.sender][_idx]];
+        BatchWithdrawalRequest memory batchRequest = batchWithdrawalRequests[request.batchId];
+        if (batchRequest.isClaimable == false) revert Unbonding();
+        if (request.claimed == true) revert AlreadyClaimed();
+
+        request.claimed = true;
+        uint256 amountInBnb = (batchRequest.amountInBnb * request.amountInBnbX) / batchRequest.amountInBnbX;
+
+        (bool success,) = payable(msg.sender).call{ value: amountInBnb }("");
+        if (!success) revert TransferFailed();
+
+        emit ClaimedWithdrawal(msg.sender, _idx, amountInBnb);
+        return amountInBnb;
+    }
+
+    /// @notice Redelegate staked BNB from one operator to another.
+    /// @param _fromOperator The address of the operator to redelegate from.
+    /// @param _toOperator The address of the operator to redelegate to.
+    /// @param _amount The amount of BNB to redelegate.
+    /// @dev redelegate has a fee associated with it. This fee will be consumed from TVL. See fn:getRedelegationFee()
+    /// @dev redelegate doesn't have a waiting period
+    function redelegate(
+        address _fromOperator,
+        address _toOperator,
+        uint256 _amount
+    )
+        external
+        override
+        nonReentrant
+        onlyRole(MANAGER_ROLE)
+    {
+        if (!OPERATOR_REGISTRY.operatorExists(_fromOperator)) {
+            revert OperatorNotExisted();
+        }
+        if (!OPERATOR_REGISTRY.operatorExists(_toOperator)) {
+            revert OperatorNotExisted();
+        }
+
+        uint256 shares = IStakeCredit(STAKE_HUB.getValidatorCreditContract(_fromOperator)).getSharesByPooledBNB(_amount);
+        STAKE_HUB.redelegate(_fromOperator, _toOperator, shares, true);
+
+        emit Redelegated(_fromOperator, _toOperator, _amount);
+    }
+
+    /// @notice Update the ER.
+    /// @dev This function is called by the manager to update the ER.
+    function updateER() external override onlyRole(MANAGER_ROLE) {
+        uint256 totalPooledBnb = getTotalStakeAcrossAllOperators();
```

Comment:
Please set limits to ER change here?

Compare old ER to new ER and revert if that goes beyond a certain threshold.

Reply:
i believe this slippage needn't be required to be checked during delgation and undelegation ??
@dulguun-staderlabs wdyt ?

Reply:
Maybe pausing the whole contract in case of a stale ER due to above?

Reply:
yeah, seems a good idea